Branch-SDK/Branch-SDK/BranchUniversalObject.m

```diff
@@ -0,0 +1,137 @@
+//
+//  BranchUniversalObject.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BranchUniversalObject.h"
+#import "BNCError.h"
+#import "BranchConstants.h"
+
+@implementation BranchUniversalObject
+
+- (instancetype)initWithCanonicalIdentifier:(NSString *)canonicalIdentifier {
+    if (self = [super init]) {
+        self.canonicalIdentifier = canonicalIdentifier;
+    }
+    return self;
+}
+
+- (instancetype)initWithTitle:(NSString *)title {
+    if (self = [super init]) {
+        self.title = title;
+    }
+    return self;
+}
+
+- (void)registerView {
+    Branch *branch = [Branch getCurrentInstanceIfAny];
+    if (!branch) {
```

Comment:
the difference is in that in all other cases, the developer has specified `getInstance` or `getTestInstance` -- with BranchUniversalObject we have to decide whether to use the test or live key. So `getCurrentInstanceIfAny` make sure it's invoked. This is different from checking whether the SDK is initialized. As long as the synchronous call to `getInstance` or `getTestInstance` has taken place, we'll know whether to use the test or live key. Maybe we should walk through this in person.

Also, a side note, the content indexing wrapper currently has this problem but blindly ignored it (I didn't think through this issue at the time). It always uses `getInstance` even if the dev specified `getTestInstance` in `application:didFinishLaunchingWithOptions:`